[GB][E2E] Add two new GB build types for AT-specific E2E testing

fullofcaffeine opened the pull request:
Project: p9oQ9f-13V-p2.

#### Changes proposed in this Pull Request

* Add two new TC builds: `Playwright Gutenberg atomic E2E tests (desktop)` and `Playwright Gutenberg atomic E2E tests (mobile)`

#### Testing instructions

* Merge and 🤞🏻 

Related to #

fullofcaffeine committed:
Add two new GB build types for AT-specific E2E testing

fullofcaffeine committed:
Missing env prefix

fullofcaffeine commented on .teamcity/_self/projects/WPComTests.kt:
```diff
@@ -214,12 +219,14 @@ fun gutenbergBuildType(screenSize: String, buildUuid: String): BuildType {
+
     return E2EBuildType (
-		buildId = "WPComTests_gutenberg_Playwright_$targetDevice",
+		buildId = "WPComTests_gutenberg_Playwright_${isAtomicDescription}_$targetDevice",
```
This actually won't work well. It'll generate `WPComTests_gutenberg_Playwright__desktop` if not-atomic. I'll fix it. The `buildId` shouldn't change for non-atomic or it might create a new entry in TC's database (AFAIK).

fullofcaffeine committed:
Keep the old buildId for the non-atomic builds

fullofcaffeine committed:
Pass actual uuids to the `buildUuid` parameter for the new builds

worldomonation commented on .teamcity/_self/projects/WPComTests.kt:
```diff
@@ -214,12 +219,23 @@ fun gutenbergBuildType(screenSize: String, buildUuid: String): BuildType {
+		// buildId should stay the same for non-atomic or TC will not find the entry
+		// and will create a new one. We want to keep all the data/history for it, so
+		// we keep the id as is.
+		"WPComTests_gutenberg_Playwright_$targetDevice"
```
I think you can change this string and it _should_ maintain history, because the UUID is the same.

I feel like we can lose the execution history for WPCOM/Gutenberg Playwright build configs, since it's just us that's looking at it at the moment. Let's try changing this string and see if the history gets kept - and we can update the FG page based on the outcome.

fullofcaffeine replied:
Sounds good! 👍🏻

fullofcaffeine replied:
Fixed in https://github.com/Automattic/wp-calypso/pull/62037/commits/8998c82ac9f16cf030a3043def66a200b77ae516

worldomonation commented on .teamcity/_self/projects/WPComTests.kt:
```diff
@@ -255,6 +271,7 @@ fun gutenbergPlaywrightBuildType( targetDevice: String, buildUuid: String ): E2E
 			)
 			param("env.AUTHENTICATE_ACCOUNTS", "gutenbergSimpleSiteEdgeUser,gutenbergSimpleSiteUser,coBlocksSimpleSiteEdgeUser,simpleSitePersonalPlanUser")
 			param("env.VIEWPORT_NAME", "$targetDevice")
+			if (atomic) Parametrized().param("env.TEST_ON_ATOMIC", "true")
```
I'm not familiar with `Parametrized`, I assume that in the case where the `if (atomic)` call evaluates to `false` the param value will not be set, is that correct?

fullofcaffeine replied:
That's what I hope will happen! But I'm not familiar with it either.

fullofcaffeine replied:
Actually, I don't need to create the instance from `Parametrized` manually here. Just calling `param` seems to be enough. I'll push a fix.

fullofcaffeine replied:
Fixed in https://github.com/Automattic/wp-calypso/pull/62037/commits/4913cb2c6626d53a914ef2a3bd6cb4b431b98adb.

worldomonation reviewed (approved):
Pre-emptively approving so you can merge it whenever you feel comfortable.

fullofcaffeine committed:
Remove conditional for `buildId` assignment

The uuid should be enough to make TC refer to the right object, so we can affort to change the `buildId`, probably :)

fullofcaffeine committed:
Simplify param expression
